pyccel/codegen/printing/fcode.py

```diff
@@ -1287,8 +1287,8 @@ def _print_NumpyAmax(self, expr):
         else:
             arg_code = self._print(array_arg)
 
-        if arg.dtype is NativeComplex():
-            test_arg = PythonReal(arg)
+        if array_arg.dtype is NativeComplex():
+            test_arg = PythonReal(array_arg)
```

Comment:
I've just realised, you probably need `NumpyReal` and `NumpyInt` instead of `PythonReal` and `PythonInt` otherwise you won't handle arrays

Reply:
It may work anyway and just have the wrong information internally though